File: src/main/java/frc/utils/controlWrappers/ArmFF.java
package frc.utils.controlWrappers;

import edu.wpi.first.math.controller.ArmFeedforward;

/*
 * wrapper for ArmFeedforward that allows setting gains from a record
 */
public class ArmFF extends ArmFeedforward{
    public ArmFF(double kS, double kG, double kV, double kA){
        super(kS, kG, kV, kA);
    }
    public ArmFF(double kS, double kG, double kV, double kA, double dt){
        super(kS, kG, kV, kA, dt);
    }
    public ArmFF(PIDGains.GravityFF gains){
        super(gains.kS(), gains.kG(), gains.kV(), gains.kA());
    }
}

File: src/main/kotlin/frc/utils/controlWrappers/ArmFF.kt
package frc.utils.controlWrappers

import edu.wpi.first.math.controller.ArmFeedforward

/*
 * wrapper for ArmFeedforward that allows setting gains from a record
 */
class ArmFF : ArmFeedforward {
    constructor(kS: Double, kG: Double, kV: Double, kA: Double) : super(kS, kG, kV, kA)
    constructor(kS: Double, kG: Double, kV: Double, kA: Double, dt: Double) : super(kS, kG, kV, kA, dt)
    constructor(gains: PIDGains.GravityFF) : super(gains.kS, gains.kG, gains.kV, gains.kA)
}
